RussianDollEnvelopes.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Time Complexity: O(N)
+ * SC = O(N)
+ * 
+ * Apporach : This is similar to longest increasing subsequence. We sort the inp array and check the longest 
```

Comment:
Please include an optimized solution as well. This one gives a time limit exceeding error.